mne/viz/topomap.py

```diff
@@ -2382,3 +2382,158 @@ def _plot_corrmap(data, subjs, indices, ch_type, ica, label, show, outlines,
     fig.canvas.draw()
     plt_show(show)
     return fig
+
+
+def _trigradient(x, y, z):
+    """Take gradients of z on a mesh."""
+    from matplotlib.tri import CubicTriInterpolator, Triangulation
+    tri = Triangulation(x, y)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        tci = CubicTriInterpolator(tri, z)
+    dx, dy = tci.gradient(tri.x, tri.y)
+    return dx, dy
+
+
+def plot_arrowmap(data, info_from, info_to=None, scale=1e-10, vmin=None,
+                  vmax=None, cmap=None, sensors=True, res=64, axes=None,
+                  names=None, show_names=False, mask=None, mask_params=None,
+                  outlines='head', contours=6, image_interp='bilinear',
+                  show=True, head_pos=None, onselect=None):
+    """Plot arrow map.
```

Comment:
sorry to give you feedback in pieces but can you also add here the ref to David's paper.

As done here:

https://github.com/mne-tools/mne-python/blob/master/mne/inverse_sparse/mxne_inverse.py#L363

Reply:
also below the "Plot arrow map" you can write a 4-5 lines paragraph to explain what it does and what it is useful for. That will really help to understand why one should be interested in this.